test/e2e/helpers/trigger_no_broker_test_helper.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/eventing/test/lib"
+	"knative.dev/eventing/test/lib/resources"
+)
+
+const (
+	brokerName = "notdefaultbroker"
+)
+
+// If shouldLabelNamespace is set to true this test annotates the testing namespace so that a default broker is created.
+// It then binds many triggers with different filtering patterns to the broker created by brokerCreator, and sends
+// different events to the broker's address.
+// Finally, it verifies that only the appropriate events are routed to the subscribers.
+func TestTriggerNoBroker(t *testing.T, channel string, brokerCreator BrokerCreator) {
+	client := lib.Setup(t, true)
+	defer lib.TearDown(client)
+	brokerName := strings.ToLower(channel)
+	subscriberName := name("dumper", "", "", map[string]interface{}{})
+	pod := resources.EventLoggerPod(subscriberName)
+	client.CreatePodOrFail(pod, lib.WithService(subscriberName))
+	client.CreateTriggerOrFailV1Beta1("testtrigger",
+		resources.WithSubscriberServiceRefForTriggerV1Beta1(subscriberName),
+		resources.WithBrokerV1Beta1(brokerName),
+	)
+
+	// Then make sure the trigger is marked as not ready since there's no broker.
+	err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		trigger, err := client.Eventing.EventingV1beta1().Triggers(client.Namespace).Get("testtrigger", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if ready := trigger.Status.GetTopLevelCondition(); ready != nil {
+			if ready.Status == corev1.ConditionFalse && ready.Reason == "BrokerDoesNotExist" {
```

Comment:
If the reason is passed as a parameter this test is a valid conformance test for the control-plane conformance.

Reply:
You mean, we could run multiple tests validations for failures? Yes, good point. I'm down to make that change, but I'd prefer to do that in a followup to make sure we get this error case handled into the release :)

Reply:
I mean, the broker conformance spec doesn't force using `BrokerDoesNotExist` as a reason for the trigger to not become ready if there is no broker; 
if the test has the reason as param implementors can reuse this test without changing the reason they're using (maybe they're using `BrokerNotFound` rather than `BrokerDoesNotExist`).

Reply:
Yep, that makes sense. As I mentioned above I'm all for it, but would prefer to do that in a followup? Ok with you?

Reply:
Of course :smile: